src/routes/transactions.ts

```diff
@@ -84,23 +92,28 @@ export const createSepaDirectDebit = async (req, res) => {
 
   if (person.account.balance.value < 0) {
     const directDebitReturn = {
-      ...queuedBooking,
+      ...booking,
       booking_type: BookingType.SEPA_DIRECT_DEBIT_RETURN,
       transaction_id: null,
-      return_transaction_id: queuedBooking.transaction_id,
+      return_transaction_id: booking.transaction_id,
     };
-    person.queuedBookings.push(directDebitReturn);
+    person.transactions.push(directDebitReturn);
+    const sepaDirectDebitReturn = createSepaDirectDebitReturn(
+      person,
+      directDebitReturn
+    );
+    await saveSepaDirectDebitReturn(sepaDirectDebitReturn);
     technicalPerson.transactions.push(directDebitReturn);
```

Comment:
oh wow, I assume this account contains a lot of bookings?

Reply:
should we update technical account? nobody cares about it.